ts/nni_manager/common/log.ts

```diff
@@ -1,141 +1,119 @@
+        this.log(levelNameToValue.info, 'INFO', args);
     }
 
     public warning(...args: any[]): void {
-        this.log(WARNING, args);
+        this.log(levelNameToValue.warning, 'WARNING', args);
     }
 
     public error(...args: any[]): void {
-        this.log(ERROR, args);
+        this.log(levelNameToValue.error, 'ERROR', args);
     }
 
     public critical(...args: any[]): void {
-        this.log(CRITICAL, args);
+        this.log(levelNameToValue.critical, 'CRITICAL', args);
     }
 
-    public fatal(...args: any[]): void {
-        this.log(FATAL, args);
+    protected log(levelValue: number, levelName: string, args: any[]): void {
+        if (levelValue >= levelNameToValue[globals.args.logLevel]) {
+            const msg = `[${timestamp()}] ${levelName} (${this.name}) ${formatArgs(args)}`;
+            globals.logStream.writeLine(msg);
+        }
     }
+}
 
-    private log(level: number, args: any[]): void {
-        const logFile: Writable | undefined = (global as any).logFile;
-        if (level < logLevel) {
+class RobustLogger extends Logger {
+    public readonly robust: boolean = true;
+    private errorOccurred: boolean = false;
+
+    protected log(levelValue: number, levelName: string, args: any[]): void {
+        if (this.errorOccurred) {
+            this.logAfterError(levelName, args);
             return;
         }
-
-        const zeroPad = (num: number): string => num.toString().padStart(2, '0');
-        const now = new Date();
-        const date = now.getFullYear() + '-' + zeroPad(now.getMonth() + 1) + '-' + zeroPad(now.getDate());
-        const time = zeroPad(now.getHours()) + ':' + zeroPad(now.getMinutes()) + ':' + zeroPad(now.getSeconds());
-        const datetime = date + ' ' + time;
-
-        const levelName = levelNames.has(level) ? levelNames.get(level) : level.toString();
-
-        const message = args.map(arg => (typeof arg === 'string' ? arg : util.inspect(arg))).join(' ');
-        
-        const record = `[${datetime}] ${levelName} (${this.name}) ${message}`;
-
-        if (logFile === undefined) {
-            if (!isUnitTest()) {  // be quite for unit test
-                console.log(record);
+        try {
+            if (levelValue >= levelNameToValue[globals.args.logLevel]) {
+                const msg = `[${timestamp()}] ${levelName} (${this.name}) ${formatArgs(args)}`;
+                globals.logStream.writeLineSync(msg);
             }
-        } else {
-            logFile.write(record + '\n');
+        } catch (error) {
+            this.errorOccurred = true;
+            console.error('[ERROR] Logger has stopped working:', error);
+            this.logAfterError(levelName, args);
         }
     }
-}
 
-export function getLogger(name: string = 'root'): Logger {
-    let logger = loggers.get(name);
-    if (logger === undefined) {
-        logger = new Logger(name);
-        loggers.set(name, logger);
+    private logAfterError(levelName: string, args: any[]): void {
```

Comment:
Are there any tests for `logAfterError`? Making sure this method is correct is important, as it's the bottomline.

Reply:
Added.